runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4224,7 +4224,10 @@ class INTERPRETER_CLASS
 			if (VM_VMHelpers::exceptionPending(_currentThread)) {
 				rc = GOTO_THROW_CURRENT_EXCEPTION;
 			} else {
-				I_32 result = (I_32)isFlattenableFieldFlattened(fieldID->declaringClass, fieldID->field);
+				I_32 result = false;
+				if (VM_ValueTypeHelpers::isNameOrSignatureQtype(J9ROMFIELDSHAPE_SIGNATURE(fieldID->field))) {
+					result = (I_32)isFlattenableFieldFlattened(fieldID->declaringClass, fieldID->field);
```

Comment:
Can you add doxygen comment for `isFlattenableFieldFlattened()` saying that the caller must ensure the field is a Q type ?

Reply:
There is already a comment saying "This helper assumes field is a qtype", is that good enough or should I update it to "The caller must ensure that the field is a qtype"?